### PR TITLE
Test SPI pins from config

### PR DESCRIPTION
**Pull request by 0xc0170**
Showcase how to use hw test pins.

Requires: https://github.com/ARMmbed/target-frdm-k64f-gcc/pull/17

@bogdanm @autopulated

### CODE REVIEW

File: source/spi_asynch.cpp [hunk 1]
```diff
@@ -33,13 +33,9 @@
...
-#if defined(TARGET_K64F)
-#define TEST_MOSI_PIN PTD2
-#define TEST_MISO_PIN PTD3
-#define TEST_SCLK_PIN PTD1
-#define TEST_CS_PIN   PTD0
-#else
-#error Target not supported
+#if !defined(YOTTA_CFG_HARDWARE_TEST_PINS_SPI_MISO) || !defined(YOTTA_CFG_HARDWARE_TEST_PINS_SPI_MOSI) || \
+    !defined(YOTTA_CFG_HARDWARE_TEST_PINS_SPI_SCLK) || !defined(YOTTA_CFG_HARDWARE_TEST_PINS_SPI_SSEL)
+#error This example requires a target to define hw spi test pins
```
Comment (autopulated) [inline source/spi_asynch.cpp]: suggest: "This example requires the SPI test pins to be defined. Please define the hardware.test-pins.spi_miso/mosi/sclk/ssel yotta config values."?

**Comment by 0xc0170** [comment]
@autopulated Updated based on your suggestion - to pinpoint where they should be defined

**Comment by autopulated** [comment]
LGTM

**Comment by bogdanm** [comment]
+1

**Comment by 0xc0170** [comment]
Ready for merge. Both of k64f targets now contain spi-test pins

**Pull request merged**